src/rust/engine/fs/src/snapshot.rs

```diff
@@ -388,6 +509,92 @@ mod tests {
     );
   }
 
+  #[test]
+  fn snapshot_merge_two_files() {
+    let (store, tempdir, _, digester) = setup();
+
+    let common_dir_name = "tower";
+    let common_dir = PathBuf::from(common_dir_name);
+
+    let dir = make_dir_stat(tempdir.path(), &common_dir);
+    let file1 = make_file_stat(
+      tempdir.path(),
+      &common_dir.join("roland"),
+      STR.as_bytes(),
+      false,
+    );
+    let file2 = make_file_stat(
+      tempdir.path(),
+      &common_dir.join("susannah"),
+      STR.as_bytes(),
+      true,
+    );
+
+    let merged = {
+      let snapshot1 =
+        Snapshot::from_path_stats(store.clone(), digester.clone(), vec![dir.clone(), file1])
+          .wait()
+          .unwrap();
+      let snapshot2 = Snapshot::from_path_stats(store.clone(), digester, vec![dir, file2])
+        .wait()
+        .unwrap();
+      Snapshot::merge(store.clone(), &[snapshot1, snapshot2])
+        .wait()
+        .unwrap()
+    };
+    let merged_root_directory = store.load_directory(merged.digest).wait().unwrap().unwrap();
+
+    assert_eq!(merged.path_stats.len(), 3);
```

Comment:
Can you compare the actual contents, rather than just the length? Ordering is well defined here

Reply:
Yea, good point.